tcf_website/views/search.py

```diff
@@ -208,22 +225,29 @@ def filter_courses(filters):
 
     return courses
 
+
 def apply_filters(results, filters):
-    """Apply filters to course queryset."""
+    """Apply filters to course queryset using Q objects for more readable code."""
+    filter_conditions = Q()
+
     if filters.get("disciplines"):
-        results = results.filter(disciplines__name__in=filters.get("disciplines"))
+        filter_conditions &= Q(disciplines__name__in=filters.get("disciplines"))
 
     if filters.get("subdepartments"):
-        results = results.filter(subdepartment__mnemonic__in=filters.get("subdepartments"))
+        filter_conditions &= Q(subdepartment__mnemonic__in=filters.get("subdepartments"))
 
     if filters.get("instructors"):
-        results = results.filter(section__instructors__id__in=filters.get("instructors"))
+        filter_conditions &= Q(section__instructors__id__in=filters.get("instructors"))
+
+    if filter_conditions:
+        results = results.filter(filter_conditions)
 
+    # Handle time filters
     weekdays = [day for day in filters.get("weekdays", []) if day]
     from_time = filters.get("from_time")
     to_time = filters.get("to_time")
 
-    if len(weekdays) != 5 and len(weekdays) != 0 or from_time or to_time:
+    if (len(weekdays) != 5 and len(weekdays) != 0) or from_time or to_time:
```

Comment:
remove superfluous parens (opinion)

Reply:
prefer `weekdays` over `len(weekdays) != 0)`, which should be `len(weekdays > 0)` regardless